crates/key-value-store/src/api/store.rs

```diff
@@ -0,0 +1,29 @@
+use async_trait::async_trait;
+use serde::{de::DeserializeOwned, Serialize};
+
+use crate::api::error::KeyValueError;
+
+#[async_trait]
+pub trait KeyValueStore<K: Clone + Sync + Send>: Clone + Sync + Send {
```

Comment:
Why does it have to be Clone?

Reply:
I'm not sure it does, I need to have a pass over some of these requirements -- not just here. I've had some "fun" with passing this type around the plugin engine with async, and to Lua, so there's been some iterations.

Reply:
Done.